crates/env/src/engine/on_chain/impls.rs

```diff
@@ -260,9 +287,10 @@ impl EnvBackend for EnvInstance {
     where
         R: scale::Encode,
     {
-        let mut scope = self.scoped_buffer();
-        let enc_return_value = scope.take_encoded(return_value);
-        ext::return_value(flags, enc_return_value);
+        let mut scope = EncodeScope::from(&mut self.buffer[..]);
+        return_value.encode_to(&mut scope);
+        let len = scope.len();
+        ext::return_value(flags, &self.buffer[..][..len]);
```

Comment:
How is this change related to this PR?

Reply:
It is related to the PR because it removed the usage of `scale::Encode::encode_to` in the WASM binary of some examples caused by storage refactoring. It can be moved to a separate PR, of course, and it still will reduce the size.

For me, that change equally can be here, or in a separate PR=) The final decision is yours

Reply:
Okay it's fine to keep it.